packages/ra-core/src/dataProvider/useMutation.ts

```diff
@@ -2,7 +2,7 @@ import { useCallback } from 'react';
 import merge from 'lodash/merge';
 
 import { useSafeSetState } from '../util/hooks';
-import useDataProvider from './useDataProvider';
+import useDataProvider from './useDataProviderWithDeclarativeSideEffects';
```

Comment:
I don't want that `useMutation` accept declarative side effects. It's a new hook, we can enforce its syntax. I just want it for `<Mutation>`.